app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1172,8 +1172,14 @@ void displayMessage(String message, boolean bad, boolean suppressToast) {
     }
 
     public static boolean isDemoUser() {
-        User u = CommCareApplication._().getSession().getLoggedInUser();
-        return (User.TYPE_DEMO.equals(u.getUserType()));
+        try {
+            User u = CommCareApplication._().getSession().getLoggedInUser();
+            return (User.TYPE_DEMO.equals(u.getUserType()));
+        } catch (SessionUnavailableException e) {
+            // Default to a normal user: this should only happen if session
```

Comment:
Shouldn't we redirect to login in this case? Is there the possibility a Demo user could end up with some functionality they shouldn't and end up crashing?